### ISSUE TITLE
# Mute Audio

### ISSUE THREAD

**Issue by jaybkim1** [begeekmyfriend/yasea#362]
This is not an issue with this awesome library. This is more of asking a question.

I will close it later. Could you generously help me to implement mute function?

I am trying to make a button where user can click to make audio sound mute.

In SrsPublisher.java, I used stopAudio() method when the button is clicked, and
the client's video view is stopped. How do you think I should do?

```
public void stopAudio() {
        if (aworker != null) {
            aworker.interrupt();
            try {
                aworker.join();
            } catch (InterruptedException e) {
                aworker.interrupt();
            }
            aworker = null;
        }

        if (mic != null) {
            mic.setRecordPositionUpdateListener(null);
            mic.stop();
            mic.release();
            mic = null;
        }

        if (aec != null) {
            aec.setEnabled(false);
            aec.release();
            aec = null;
        }

        if (agc != null) {
            agc.setEnabled(false);
            agc.release();
            agc = null;
        }
    }

```

**Comment by begeekmyfriend** [comment]
Change this [line](https://github.com/begeekmyfriend/yasea/blob/master/app/src/main/java/net/ossrs/yasea/demo/MainActivity.java#L135) to `mPublisher.stopAudio` and another branch to `mPublisher.startAudio`. And do not forget to comment out this [line](https://github.com/begeekmyfriend/yasea/blob/master/app/src/main/java/net/ossrs/yasea/demo/MainActivity.java#L96).

**Comment by begeekmyfriend** [comment]
I see, some flash player has to work after receiving ADTS header, so solution #344 maybe good for you.

**Comment by jaybkim1** [comment]
Forgot to comment.
Thank you !
I solved it from #344
